pandas/_libs/src/parser/tokenizer.c

```diff
@@ -1148,13 +1160,17 @@ static int parser_handle_eof(parser_t *self) {
 
         case ESCAPE_IN_QUOTED_FIELD:
         case IN_QUOTED_FIELD:
+            if (!self->error_bad_lines)
+                return 0;
```

Comment:
If statements without braces are a common source of bugs, especially related to version control merging. I would prefer to see braces on all if statements.